src/main/java/org/simplify4u/plugins/PGPVerifyMojo.java

```diff
@@ -467,9 +467,9 @@ private boolean verifyPGPSignature(Artifact artifact, Artifact ascArtifact) thro
                     + " signature.hashAlgorithm: " + pgpSignature.getHashAlgorithm());
 
             return verifySignatureStatus(pgpSignature.verify(), artifact, publicKey, publicKeyRing);
-        } catch (IOException | PGPException e) {
-            if (e.getCause() instanceof PGPKeyNotFound && keysMap.isKeyMissing(artifact)) {
-                final String logMessage = String.format("%s PGP Key not found on server, consistent with keys map.",
+        } catch (PGPKeyNotFound e) {
```

Comment:
I'm just curious - previously the type of the `cause` was checked, and now the type of actual caught object. Is there any test to show that the previous code worked at all - and to check that after this change it works as well?

Reply:
Now that you mention it, I believe there is no test. I'll see if I can add one. I only tested on a real code-base now.

Reply:
@pzygielo okay, there does not seem to be a test, currently, that confirms the `noKey` specification, i.e. accepting missing public key. It might be best to publish a "test" module signed by a deleted key that can then be used as part of an Integration Test. In that case I might be best to publish as part of _github.com/s4u_.

Reply:
Those lines of code have no coverage by test:
https://sonarcloud.io/code?id=org.simplify4u.plugins%3Apgpverify-maven-plugin&selected=org.simplify4u.plugins%3Apgpverify-maven-plugin%3Asrc%2Fmain%2Fjava%2Forg%2Fsimplify4u%2Fplugins%2FPGPVerifyMojo.java&line=470

Reply:
Adding fake artifact signed by key which not exist in public key servers maybe not possible because sonatype verify if key is deployed to key servers.

But we can add it to `src/it/resources/local-repo` - it is start content for  local repository used during test.

Reply:
2 IT tests added: one testing `noKey` in keysmap, one testing failure to acquire public key.